### PR TITLE
refactor(bundlr): add file type to end of image and animation URLs

### LINKED ISSUE #303: [Bug]: sugar upload not appending the ext when using Bundlr
### Issue description

The command sugar upload doesn't append the proper extension to images/animations.
It is problematic for animations as various wallets won't display the video correctly.

My assets folder looks like this:
Screenshot 2022-08-01 at 21 53 05

This is the JSON file for one of the NFT with the animation_url.
[Token Metadata Standard](https://docs.metaplex.com/programs/token-metadata/token-standard) is respected.
```json
{
    "name": "DEMO BUG",
    "symbol": "BUG",
    "description": "Some description",
    "image": "0.png",
    "animation_url": "0.mp4",
    "external_url": "",
    "attributes": [
        {
          "trait_type": "Item",
          "value": "xxx"
        }
      ],
    "properties": {
      "category": "video",
      "files": [
        {
          "uri": "0.png",
          "type": "image/png"
        },
        {
          "uri": "0.mp4",
          "type": "video/mp4"
        }
      ]
    }
  }
```

After `sugar upload` check **cache.json** and open **metadata_link** for one of the NFT:

```json
{
  "name": "ULTRA GALACTIC ESTHER",
  "symbol": "",
  "description": "Esther stumbles into a peculiar green hue, falling, and finds herself at the Mortuary doorstep. Upon entering, Esther feast her eyes on the tombs in front of her, struck at the opportunity to plunder, loot and play dress-ups.",
  "image": "https://arweave.net/32M4ARiDKDwjLfukYKC378O6FPeWnXnpHh9G2oSpI20",
  "animation_url": "https://arweave.net/yLCKZAAO4GgHhJ8ShhtCpc0g0cVwrBdA3mz7r5enVC8",
  "external_url": "https://www.esthersescape.xyz/esthers-mortuary-plunder",
  "attributes": [
    {
      "trait_type": "ITEM #1",
      "value": "JETPACK"
    },
    {
      "trait_type": "ITEM #2",
      "value": "GOGGLES"
    },
    {
      "trait_type": "ITEM #3",
      "value": "BLASTER CANNON"
    },
    {
      "trait_type": "EYE COUNT",
      "value": "5"
    },
    {
      "trait_type": "STANCE",
      "value": "ANIMATED"
    },
    {
      "trait_type": "ABILITY",
      "value": "TRANSPLANAR GLIDE"
    },
    {
      "trait_type": "RARITY",
      "value": "ULTRA"
    }
  ],
  "properties": {
    "files": [
      {
        "uri": "https://arweave.net/32M4ARiDKDwjLfukYKC378O6FPeWnXnpHh9G2oSpI20",
        "type": "image/png"
      },
      {
        "uri": "https://arweave.net/yLCKZAAO4GgHhJ8ShhtCpc0g0cVwrBdA3mz7r5enVC8",
        "type": "video/mp4"
      }
    ]
  }
}
```

### Relevant log output

_No response_

### Priority this issue should have

Medium (should be fixed soon)

### PR DESCRIPTION
**Pull request by samuelvanderwaal**
Images and animation URLs will now look like:

```
https://arweave.net/5UnllZT23t5WTqd61Ho4_-J2GxZihEBSdFkZD9d8KI0?ext=png
```

Fixes #303.

### CODE REVIEW

File: src/upload/methods/bundlr.rs [hunk 1]
```diff
@@ -197,7 +197,20 @@ impl BundlrMethod {
...
-        let link = format!("https://arweave.net/{}", id);
+        // Get extension for the asset type.
+        let ext = asset_info
+            .content_type
+            .split('/')
+            .nth(1)
+            .ok_or_else(|| anyhow!("Failed context type to get extension"))?;
+
+        let link = match asset_info.data_type {
+            DataType::Image => format!("https://arweave.net/{id}?ext={ext}"),
```
Comment (febo) [inline src/upload/methods/bundlr.rs:208]: Would it make sense to simplify the code of the match? E.g.:
```
let link = match asset_info.data_type {
    DataType::Image | DataType::Animation => {
        format!("https://arweave.net/{id}?ext={ext}")
     },
    DataType::Metadata => format!("https://arweave.net/{}", id),
}
```

File: src/upload/methods/bundlr.rs [hunk 2]
```diff
@@ -197,7 +197,20 @@ impl BundlrMethod {
...
+            DataType::Metadata => format!("https://arweave.net/{}", id),
```
Comment (febo) [inline src/upload/methods/bundlr.rs:209]: `"https://arweave.net/{id}"` for consistency?

File: src/upload/methods/bundlr.rs (hunk 1 above)
Comment (samuelvanderwaal) [reply src/upload/methods/bundlr.rs:208]: yeah, totally, will fix

File: src/upload/methods/bundlr.rs (hunk 2 above)
Comment (samuelvanderwaal) [reply src/upload/methods/bundlr.rs:209]: good call, will fix

**Review by febo — APPROVED** [review]

**Pull request merged**